ompi/mca/coll/basic/coll_basic_module.c

```diff
@@ -184,6 +181,12 @@ int
 mca_coll_basic_module_enable(mca_coll_base_module_t *module,
                              struct ompi_communicator_t *comm)
 {
+    /* prepare the placeholder for the array of request* */
+    module->base_data = OBJ_NEW(mca_coll_basic_comm_t);
```

Comment:
@derbeyn now I see what you said the coll_basic_comm_t is useless. Indeed it can be replaced.